relayer-cli/src/commands/clear.rs

```diff
@@ -0,0 +1,98 @@
+use abscissa_core::clap::Parser;
+use abscissa_core::{Command, Runnable};
+
+use ibc::core::ics24_host::identifier::{ChainId, ChannelId, PortId};
+use ibc::events::IbcEvent;
+use ibc_relayer::link::error::LinkError;
+use ibc_relayer::link::{Link, LinkParameters};
+
+use crate::application::app_config;
+use crate::cli_utils::ChainHandlePair;
+use crate::conclude::Output;
+use crate::error::Error;
+
+/// `clear` subcommands
+#[derive(Command, Debug, Parser, Runnable)]
+pub enum ClearCmds {
+    /// Clear outstanding packets (i.e., packet-recv and packet-ack)
+    /// on a given channel in both directions.
+    Packets(ClearPacketsCmd),
+}
+
+#[derive(Debug, Parser)]
+pub struct ClearPacketsCmd {
+    #[clap(required = true, help = "identifier of the destination chain")]
+    dst_chain_id: ChainId,
+
+    #[clap(required = true, help = "identifier of the source chain")]
+    src_chain_id: ChainId,
+
+    #[clap(required = true, help = "identifier of the source port")]
+    src_port_id: PortId,
+
+    #[clap(required = true, help = "identifier of the source channel")]
+    src_channel_id: ChannelId,
+}
+
+impl Runnable for ClearPacketsCmd {
+    fn run(&self) {
+        let config = app_config();
+
+        let chains = match ChainHandlePair::spawn(&config, &self.src_chain_id, &self.dst_chain_id) {
+            Ok(chains) => chains,
+            Err(e) => Output::error(format!("{}", e)).exit(),
+        };
+
+        let mut ev_list = vec![];
+
+        // Construct links in both directions.
+        // Some of the checks in the two Link constructor calls may be redundant;
+        // going slowly, but reliably.
+        let opts = LinkParameters {
+            src_port_id: self.src_port_id.clone(),
+            src_channel_id: self.src_channel_id.clone(),
+        };
+        let fwd_link =
+            match Link::new_from_opts(chains.src.clone(), chains.dst.clone(), opts, false) {
+                Ok(link) => link,
+                Err(e) => Output::error(format!("{}", e)).exit(),
+            };
+        let opts = LinkParameters {
+            src_port_id: fwd_link.a_to_b.dst_port_id().clone(),
+            src_channel_id: fwd_link.a_to_b.dst_channel_id().clone(),
+        };
+        let rev_link = match Link::new_from_opts(chains.dst, chains.src, opts, false) {
+            Ok(link) => link,
+            Err(e) => Output::error(format!("{}", e)).exit(),
+        };
```

Comment:
This may be the case for a `Link::reverse` method, which would do this and maybe skip on some unnecessary checks.

Reply:
Sounds like a nice addition, would you mind doing that as part of this PR?